### PR TITLE
Added a filter input to the queue menu

### PR DESCRIPTION
**Pull request by Jivings**
I have a lot of queues to manage in my bull-board dashboard (around 200-300 queues). I know the names of each of them but because of the truncation in the sidebar menu I can't just `cmd+f` search for the one I'm looking for.

So I added this little change to the Menu that allows me to filter the queues that appear on the left so I can find the one I'm looking for.

image

image

Let me know what you think, happy to tweak the UX if you have something else in mind.

Cheers!

### CODE REVIEW

File: packages/ui/src/components/Menu/Menu.module.css [hunk 1]
```diff
@@ -1,72 +1,106 @@
...
-    font-size: 0.833em;
-   display: block;
-margin-bottom: -0.75em;
+  color: #828e97;
+  font-size: 0.833em;
+  display: block;
+  margin-bottom: -0.75em;
+}
+
+.search-wrapper {
```
Comment (felixmosh) [inline packages/ui/src/components/Menu/Menu.module.css:74]: please use the same convention (`camelCase` for selectors)

File: packages/ui/src/components/Menu/Menu.module.css [hunk 2]
```diff
@@ -1,72 +1,106 @@
...
+  padding-top: 1rem;
+  color: #d5d9dc;
+  display: flex;
+  flex-direction: column;
+  box-shadow: 4px 0 8px 3px rgba(38, 46, 60, 0.1);
 }
 
 .aside > div {
-    color: #828e97;
-    font-size: 0.833em;
```
Comment (felixmosh) [inline packages/ui/src/components/Menu/Menu.module.css:23]: please enable prettier at you coding env, so it will align styling

**Review by felixmosh — CHANGES_REQUESTED** [review]
Thank you for the PR 🙏🏼

It looks really great.

File: packages/ui/src/components/Menu/Menu.tsx [hunk 3]
```diff
@@ -11,30 +12,53 @@ export const Menu = ({
...
-  </aside>
-);
+}) => {
+  const [searchTerm, setSearchTerm] = useState('');
+  return (
+    <aside className={s.aside}>
+      <div>QUEUES</div>
+
+      <span className={s.searchWrapper}>
+        <svg className={s.searchIcon} xmlns="http://www.w3.org/2000/svg"
```
Comment (felixmosh) [inline packages/ui/src/components/Menu/Menu.tsx:22]: Make the Icon as rest of the icons (a React component)

**Review by felixmosh — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/Menu/Menu.module.css (hunk 2 above)
Comment (Jivings) [reply packages/ui/src/components/Menu/Menu.module.css:23]: Hm, my prettier auto-formatted the file to change the 4 spaces to 2. Not sure why.

Comment (Jivings) [reply packages/ui/src/components/Menu/Menu.module.css:23]: I think there's no tabWidth set specifically for CSS files in the `prettier.config.js`, do you have local config for that?

Comment (felixmosh) [reply packages/ui/src/components/Menu/Menu.module.css:23]: Your are correct, this file was missed out...

**Comment by felixmosh** [comment]
I will tweek the UI a bit so it will fit more, and then publish it.
OK?

**Review by felixmosh — APPROVED** [review]

**Comment by Jivings** [comment]
Yep sounds good to me!

**Comment by Jivings** [comment]
I pushed the fixes you requested too, don't miss those commits.

**Comment by felixmosh** [comment]
Final result
image

Thank you for this feature 🙏🏼

**Comment by Jivings** [comment]
Thanks for the quick turnaround! Looks great!

**Pull request merged**